yt_dlp/downloader/http.py

```diff
@@ -196,13 +185,7 @@ def establish_connection():
                     # Unexpected HTTP error
                     raise
                 raise RetryDownload(err)
-            except urllib.error.URLError as err:
-                if isinstance(err.reason, ssl.CertificateError):
-                    raise
-                raise RetryDownload(err)
-            # In urllib.request.AbstractHTTPHandler, the response is partially read on request.
-            # Any errors that occur during this will not be wrapped by URLError
-            except RESPONSE_READ_EXCEPTIONS as err:
+            except TransportError as err:
```

Comment:
We are excluding certificate error here. (might not be possible to easily differentiate though)

Reply:
Yeah we can have a CertificateError extend from SSLError, but as you say would require the backends to differentiate that. I want to try limit how many errors the backends have to differentiate as that can be difficult.

Reply:
Added `CertificateVerifyError`. It shouldn't be too difficult to differentiate this particular error from other SSL errors.